crates/circuit/src/circuit_instruction.rs

```diff
@@ -459,10 +715,24 @@ impl CircuitInstruction {
 #[derive(Debug)]
 pub struct OperationFromPython {
     pub operation: PackedOperation,
-    pub params: SmallVec<[Param; 3]>,
+    pub params: Option<Parameters<PyObject>>,
     pub label: Option<Box<String>>,
 }
```

Comment:
Not for this PR, but would it make sense to rename this to `InstructionFromPython`? Or would it need to contain qargs and/or cargs for it to count.